crates/sui-types/src/transaction.rs

```diff
@@ -2349,14 +2371,25 @@ impl VerifiedTransaction {
         epoch: u64,
         round: u64,
         commit_timestamp_ms: CheckpointTimestamp,
+        consensus_digest: Option<ConsensusCommitDigest>,
```

Comment:
No strong opinion, but I think it might look clearer if we have two separate constructors, for v1 and v2 respectively, instead of relying on the None-ness of the argument.

Reply:
Done. I didn't separate the two constructors because we need to make multiple `v2` functions from the call site to here. But I can also see the benefit to separate V1 and V2 from the call site of generating the consensus prologue transaction.